client/my-sites/plan-features/index.jsx

```diff
@@ -139,6 +141,69 @@ class PlanFeatures extends Component {
 		);
 	}
 
+	hasDiscountNotice() {
+		const { canPurchase, hasPlaceholders, withDiscount } = this.props;
+		const bannerContainer = this.getBannerContainer();
+		if ( ! bannerContainer ) {
+			return false;
+		}
+
+		const activeDiscount = getDiscountByName( withDiscount );
+		if ( ! activeDiscount || hasPlaceholders || ! canPurchase ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	renderCreditNotice() {
+		const {
+			canPurchase,
+			hasPlaceholders,
+			translate,
+			planCredits,
+			planProperties,
+			showPlanCreditsApplied,
+		} = this.props;
+		const bannerContainer = this.getBannerContainer();
+		if (
+			hasPlaceholders ||
+			! canPurchase ||
+			! bannerContainer ||
+			! showPlanCreditsApplied ||
+			! planCredits
+		) {
+			return null;
+		}
+
+		return ReactDOM.createPortal(
```

Comment:
:weary: Please no Portals -- using them to just 'get a component in place' somewhere, they are the UI equivalent of :spaghetti: code. See prior discussion [here](https://github.com/Automattic/wp-calypso/pull/22384#issuecomment-382773722).

Reply:
@ockham interesting! Portals are already used to display another notice from the same component and I simply copied that. I don't agree we should avoid Portals for all cost, although in case of this component we could totally rid of them in a short janitorial PR.